redis/connection.py

```diff
@@ -124,9 +124,12 @@ def encode(self, value):
         return value
 
     def decode(self, value, force=False):
-        "Return a unicode string from the byte representation"
-        if (self.decode_responses or force) and isinstance(value, bytes):
-            value = value.decode(self.encoding, self.encoding_errors)
+        "Return a unicode string from the bytes-like representation"
+        if (self.decode_responses or force):
+            if isinstance(value, memoryview):
```

Comment:
Did you run into a case where you needed this? Are we ever actually decoding `memoryview` instances?

Reply:
I don't think I ran into a case in the tests where a `memoryview` gets decoded.  I think I got a bit spooked by lines like [this](https://github.com/andymccurdy/redis-py/blob/4f0eed6c7b8cc2ad02eed71b3097b8f07a87ee84/redis/client.py#L1083) and [this](https://github.com/andymccurdy/redis-py/blob/4f0eed6c7b8cc2ad02eed71b3097b8f07a87ee84/redis/client.py#L1090), but on a closer look those lines shouldn't cause a problem unless the user passes a `memoryview` kwarg -- and the docs instruct them to pass strings.  Maybe it just felt right to make sure that `decode(encode(x))` always works, but it seems it's not essential, happy to remove this if you prefer.